src/jobs/jobs.controller.ts

```diff
@@ -356,16 +462,16 @@ export class JobsController {
           HttpStatus.BAD_REQUEST,
         );
       }
-    } else if (jobParams.auth == JobsAuth.DatasetAccess) {
+    } else if (jc[functionType].auth == JobsAuth.DatasetAccess) {
       // verify that all the pids listed in the property indicated are accessible by the user
-      const datasetIds = this.checkDatasetIds(jobParams);
+      const datasetIds = this.checkDatasetIds(createJobDto.jobParams);
       const numberOfDatasets = await this.datasetsService.count({
         where: {
           pid: { $in: datasetIds },
           $or: [
             { ownerGroup: { $in: user.currentGroups } },
```

Comment:
Doesn't this give an error for anonymous users (`user === undefined`)?

Reply:
Adding one extra function to verify user is authenticated